cluster/points_writer.go

```diff
@@ -196,6 +202,13 @@ func (w *PointsWriter) WritePoints(database, retentionPolicy string, consistency
 	w.statMap.Add(statWriteReq, 1)
 	w.statMap.Add(statPointWriteReq, int64(len(points)))
 
+	if w.MaxConcurrentWrites > 0 {
```

Comment:
As I understand it, the issue is with the underlying TCP connections in HTTP, and not really a problem with limiting the points written internally, #6559 is specific to HTTP.

This also doesn't solve the problem of new connections being created. They will still be created, and then return an error, rather than closing the connection immediately when the limit has been exceeded. All this does (including `startWrite`), is add further points of lock contention in the write path...